packages/composite-checkout-wpcom/test/translate-cart.js

```diff
@@ -323,4 +326,221 @@ describe( 'translateWpcomCartToCheckoutCart', function() {
+					is_bundled: true,
+					item_subtotal: 0,
+					item_subtotal_integer: 0,
+					item_subtotal_display: '$0',
+					item_tax: 0,
+					item_total: 0,
+					subscription_id: 0,
+				},
+				{
+					product_id: 69,
+					product_name: 'G Suite',
+					product_name_en: 'G Suite',
+					product_slug: 'gapps',
+					product_cost: 72,
+					meta: 'foo.cash',
+					cost: 72,
+					currency: 'USD',
+					volume: 2,
+					free_trial: false,
+					orig_cost: null,
+					cost_before_coupon: null,
+					extra: {
+						context: 'signup',
+						google_apps_users: [
+							{
+								email: 'foo@foo.cash',
+								firstname: 'First',
+								lastname: 'User',
+							},
+							{
+								email: 'bar@foo.cash',
+								firstname: 'Second',
+								lastname: 'User',
+							},
+						],
+					},
+					bill_period: '365',
+					is_domain_registration: false,
+					time_added_to_cart: 1572551402,
+					is_bundled: false,
+					item_subtotal: 72,
+					item_subtotal_integer: 7200,
+					item_subtotal_display: '$72',
+					item_tax: 0,
+					item_total: 72,
+					subscription_id: 0,
+				},
+			],
+			tax: [],
+			sub_total: '216',
+			sub_total_display: '$216',
+			sub_total_integer: 21600,
+			total_tax: '5',
+			total_tax_display: '$5',
+			total_tax_integer: 500,
+			total_cost: 221,
+			total_cost_display: '$221',
+			total_cost_integer: 22100,
+			currency: 'USD',
+			allowed_payment_methods: [
+				'WPCOM_Billing_Stripe_Payment_Method',
+				'WPCOM_Billing_Ebanx',
+				'WPCOM_Billing_Web_Payment',
+			],
+		};
+
+		const clientCart = translateWpcomCartToCheckoutCart( serverResponse );
+
+		it( 'has a total property', function() {
+			expect( clientCart.total.amount ).toBeDefined();
+		} );
+		it( 'has the expected total value', function() {
+			expect( clientCart.total.amount.value ).toBe( 22100 );
+		} );
+		it( 'has the expected currency', function() {
+			expect( clientCart.total.amount.currency ).toBe( 'USD' );
+		} );
+		it( 'has the expected total display value', function() {
+			expect( clientCart.total.amount.displayValue ).toBe( '$221' );
+		} );
+		it( 'has a list of items', function() {
+			expect( clientCart.items ).toBeDefined();
+		} );
+		it( 'has the expected number of line items', function() {
+			expect( clientCart.items.length ).toBe( 3 );
+		} );
+		it( 'has an array of allowed payment methods', function() {
+			expect( clientCart.allowedPaymentMethods ).toBeDefined();
+		} );
+
+		describe( 'third cart item (GSuite)', function() {
+			it( 'has an id', function() {
+				expect( clientCart.items[ 2 ].id ).toBeDefined();
+			} );
+			it( 'has the expected label', function() {
+				expect( clientCart.items[ 2 ].label ).toBe( 'G Suite' );
+			} );
+			it( 'has the expected product_id', function() {
+				expect( clientCart.items[ 2 ].wpcom_meta?.product_id ).toBe( 69 );
+			} );
+			it( 'has the expected google_apps_users', function() {
+				expect( clientCart.items[ 2 ].wpcom_meta?.extra?.google_apps_users?.[ 0 ] ).toEqual( {
```

Comment:
We don't, really. The test will fail either way if the data is not there. I think I did this because without the change in the code, the expression `clientCart.items[2].wpcom_meta.extra` would cause a javascript Error to be thrown since `extra` is not a property on `wpcom_meta`. 

I think, but have not tested, that this would give a different failure message when the test was run, and that message might not be as clear.

With optional chaining, no javascript Error will be thrown, and the `expect` will simply fail to equal the expected value, which I imagine is a more helpful error. 🤷‍♂